`--site-isolation` isn't running tests in a cross-origin iframe

charliewolfe opened the pull request:
#### be13d892218cf6b970dc28a19e5b16042924ed62
<pre>
`--site-isolation` isn&apos;t running tests in a cross-origin iframe
<a href="https://bugs.webkit.org/show_bug.cgi?id=270114">https://bugs.webkit.org/show_bug.cgi?id=270114</a>
<a href="https://rdar.apple.com/123646865">rdar://123646865</a>

Reviewed by Alex Christensen.

A typo was preventing it from being enabled.

Most tests actually fail, so only enable a few for now.

* LayoutTests/platform/mac-site-isolation/TestExpectations:
* Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
(_set_up_derived_options):
* Tools/Scripts/webkitpy/port/driver.py:
(Driver.is_http_test):
* Tools/Scripts/webkitpy/port/driver_unittest.py:
* Tools/WebKitTestRunner/TestInvocation.cpp:
(WTR::TestInvocation::invoke):
* Tools/WebKitTestRunner/TestOptions.cpp:
(WTR::TestOptions::defaults):
(WTR::TestOptions::keyTypeMapping):
* Tools/WebKitTestRunner/TestOptions.h:
(WTR::TestOptions::runInCrossOriginFrame const):
(WTR::TestOptions::runInCrossOriginIFrame const): Deleted.

Canonical link: <a href="https://commits.webkit.org/275837@main">https://commits.webkit.org/275837@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/67d618edea04509ffbefe122ba7159147007d8a3

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/42945 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/21968 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/45346 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/45565 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/39070 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/45251 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/25752 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/19393 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/35528 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/43518 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/19061 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/36967 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/16505 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/42816 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/16653 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/38025 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/1001 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/39184 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/38359 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/47090 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/17780 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/14648 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/42286 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/19397 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/40946 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/19575 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5825 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/19027 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rreno commented on Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
```diff
@@ -478,7 +478,7 @@ def _set_up_derived_options(port, options):
 
     if port.port_name == "mac" and options.site_isolation:
-        options.self_compare_with_header = 'SiteIsolationEnabled=true runInCrossOriginIframe=true'
+        options.self_compare_with_header = 'SiteIsolationEnabled=true runInCrossOriginIFrame=true'
```
Can we just make this directive `runInCrossOriginFrame` (no `I`) so we don't make this typo again? I only worked on site isolation for about a week and I messed up the capitalization almost every time I tried to use the directive in an HTML comment.

rreno replied:
I should have probably made the fix myself but I thought I must be the only one making the mistake at the time.

charliewolfe replied:
Yes, we should do that before merging this.

charliewolfe committed:
`--site-isolation` isn't running tests in a cross-origin iframe
https://bugs.webkit.org/show_bug.cgi?id=270114
rdar://123646865

Reviewed by Alex Christensen.

A typo was preventing it from being enabled.

Most tests actually fail, so only enable a few for now.

* LayoutTests/platform/mac-site-isolation/TestExpectations:
* Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
(_set_up_derived_options):
* Tools/Scripts/webkitpy/port/driver.py:
(Driver.is_http_test):
* Tools/Scripts/webkitpy/port/driver_unittest.py:
* Tools/WebKitTestRunner/TestInvocation.cpp:
(WTR::TestInvocation::invoke):
* Tools/WebKitTestRunner/TestOptions.cpp:
(WTR::TestOptions::defaults):
(WTR::TestOptions::keyTypeMapping):
* Tools/WebKitTestRunner/TestOptions.h:
(WTR::TestOptions::runInCrossOriginFrame const):
(WTR::TestOptions::runInCrossOriginIFrame const): Deleted.

Canonical link: https://commits.webkit.org/275837@main
